src/Symfony/Component/Mailer/Transport/FileTransport.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Mailer\Transport;
+
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Mailer\SentMessage;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
+
+/**
+ * Pretends messages have been sent, but save them in the filesystem.
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ *
+ * @experimental in 4.3
+ */
+final class FileTransport extends AbstractTransport
+{
+    private $path;
+
+    public function __construct(string $path, EventDispatcherInterface $dispatcher = null, LoggerInterface $logger = null)
+    {
+        parent::__construct($dispatcher, $logger);
+
+        $this->path = $path;
+
+        if (!file_exists($this->path)) {
+            if (!mkdir($this->path, 0777, true)) {
+                throw new \RuntimeException(sprintf('Unable to create path "%s".', $this->path));
+            }
+        }
+    }
+
+    protected function doSend(SentMessage $message): void
+    {
+        $file = $this->path.'/'.uniqid().'.message';
+        $serializedMessage = serialize($message);
+
+        if (false === file_put_contents($file, $serializedMessage)) {
```

Comment:
Why not using Filesystem here too?

Reply:
I didn't wanted to make the Mailer component depends of Filesystem, but sure I can use it 👍

Reply:
Not sure what others think

Reply:
IMO no need of new dependency